app/src/main/java/ca/klapstein/baudit/Data/Password.java

```diff
@@ -23,8 +28,32 @@ public String getPassword() {
         return password;
     }
 
-    public void setPassword(@NonNull String password) {
-        // TODO: validate password
-        this.password = password;
+    public void setPassword(@NonNull String password) throws IllegalArgumentException {
+        Pattern p = Pattern.compile(VALID_PASSWORD);
+        Matcher m = p.matcher(password);
+        if (m.matches()){
+            this.password = password;
+        } else {
+            throw new IllegalArgumentException();
```

Comment:
should provide an exception message e.g.

```java
 throw new IllegalArgumentException("Invalid password string");
```

bonus find some way to get the invalid string to be included in this message aswell for easier debugging

Reply:
included msg